### PR TITLE
refactor(core,phrases): rm empty strings and check structure in PUT /custom-phrase

### PR DESCRIPTION
**Pull request by IceHe**
## Summary

`PUT /custom-phrase` route

- Remove empty string values in the input translation.
- If the input translation structure is invalid, it will fail.

## Testing

Locally tested

- UTs

image

- HTTP requests

image

image

### CODE REVIEW

File: packages/core/src/utils/translation.test.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+  test.each([null, true, false, 123])(
+    'should be false when there is an unexpected value `%p` in the structure',
+    (unexpectedValue) => {
+      expect(
+        isValidStructure(en.translation, {
+          secondary: {
+            sign_in_with: 'Se connecter avec {{methods, list(type: disjunction;)}}',
+            // @ts-expect-error for testing
+            social_bind_with: unexpectedValue,
```
[the remark this answers is not in the document]
Comment (IceHe) [reply packages/core/src/utils/translation.test.ts:38]: Removed the unnecessary test case.

File: packages/core/src/utils/translation.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import { Translation } from '@logto/schemas';
+
+export const isValidStructure = (
+  fullTranslation: Translation,
+  partialTranslation: Partial<Translation>
```
Comment (wangsijie) [inline packages/core/src/utils/translation.ts:5]: For the usage, you can not ensure that this parameter is `Partial<Translation>`, maybe change to a nested record of string.

**Review by simeng-li — APPROVED** [review]
LGTM

File: packages/core/src/routes/custom-phrase.test.ts [hunk 3]
```diff
@@ -39,6 +40,15 @@ jest.mock('@/queries/custom-phrase', () => ({
   upsertCustomPhrase: async (customPhrase: CustomPhrase) => upsertCustomPhrase(customPhrase),
 }));
 
+const isValidStructure = jest.fn(
+  (fullTranslation: Translation, partialTranslation: Partial<Translation>) => true
```
Comment (wangsijie) [inline packages/core/src/routes/custom-phrase.test.ts:44]: `(fullTranslation: Translation, partialTranslation: Translation)`

**Review by wangsijie — APPROVED** [review]
LGTM

**Pull request merged**